### ISSUE TITLE
[Node 18] Global path is available by default, but "path" is not defined is reported

### ISSUE THREAD

**Issue by marcbuhler** [eslint/eslint#16839]
### Environment

Node version: 18.12.1
npm version: 8.19.2
Local ESLint version: 8.32.0
Global ESLint version: 8.32.0
Operating System: Windows 10

### What parser are you using?

Default (Espree)

### What did you do?

Similarly to #15941 `path` and `fs` are reported as "not defined" though NodeJS 18 provides these by default without import or require.

### What did you expect to happen?

fs and path should be available as implicit globals

### What actually happened?

```bash
  56:22  error  'path' is not defined  no-undef
  57:12  error  'fs' is not defined    no-undef
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

### .eslintrc.json
```json
{
  "extends": "strongloop",
  "env": {
    "es6": true,
    "jest": true,
    "node": true
  },
  "parserOptions": {
    "ecmaVersion": 2020
  },
  "rules": {
    "max-len": "off",
    "no-shadow": "error",
    "quotes": [
      "error",
      "single"
    ],
    "object-curly-spacing": [
      "error",
      "never"
    ],
    "quote-props": "off",
    "block-spacing": "off",
    "space-before-function-paren": "off",
    "complexity": [
      "warn",
      10
    ],
    "require-await": "error"
  }
}
```

**Comment by nzakas** [comment]
These are pulled in from the [`globals`](https://github.com/sindresorhus/globals/) package, so we'd need that package to be updated first. Can you open an issue or pull request there?

**Comment by cawoodm** [comment]
OK, I'll start with the offical [docs](https://nodejs.org/docs/latest-v18.x/api/globals.html) which [don't](https://github.com/nodejs/node/issues/46456) mention them

**Comment by marcbuhler** [comment]
OK, I'll start with the offical [docs](https://nodejs.org/docs/latest-v18.x/api/globals.html) which [don't](https://github.com/nodejs/node/issues/46456) mention them

**Comment by aduh95** [comment]
> NodeJS 18 provides these by default without import or require

That doesn't seem to be correct:

```console
$ echo 'console.log(typeof fs, typeof path)' | node
undefined undefined
$ echo 'console.log(fs.readFile())' | node 
[stdin]:1
console.log(fs.readFile())
            ^

ReferenceError: fs is not defined
    at [stdin]:1:13
    at Script.runInThisContext (node:vm:128:12)
    at Object.runInThisContext (node:vm:306:38)
    at node:internal/process/execution:83:21
    at [stdin]-wrapper:6:24
    at runScript (node:internal/process/execution:82:62)
    at evalScript (node:internal/process/execution:104:10)
    at node:internal/main/eval_stdin:31:5
    at Socket.<anonymous> (node:internal/process/execution:205:5)
    at Socket.emit (node:events:525:35)

Node.js v19.4.0
$ echo 'console.log(path.join())' | node    
[stdin]:1
console.log(path.join())
            ^

ReferenceError: path is not defined
    at [stdin]:1:13
    at Script.runInThisContext (node:vm:128:12)
    at Object.runInThisContext (node:vm:306:38)
    at node:internal/process/execution:83:21
    at [stdin]-wrapper:6:24
    at runScript (node:internal/process/execution:82:62)
    at evalScript (node:internal/process/execution:104:10)
    at node:internal/main/eval_stdin:31:5
    at Socket.<anonymous> (node:internal/process/execution:205:5)
    at Socket.emit (node:events:525:35)

Node.js v19.4.0
```

**Comment by marcbuhler** [comment]
Weird but true ☝

However, run `node` v18/19 and then type in `typeof fs`:

![image](https://user-images.githubusercontent.com/123749354/216094914-ab66d106-4567-4ca8-87ad-4d2841d7baca.png)

**Comment by aduh95** [comment]
It seems that the screenshot you shared is a screenshot of the Node.js REPL, which by default put all built-in modules on the global scope (see https://nodejs.org/api/repl.html#accessing-core-nodejs-modules).

**Comment by marcbuhler** [comment]
OK, my bad! Thanks for clearing that up!
